### DISCUSSION TITLE
An idea about returning values from block expressions

### DISCUSSION THREAD

**Discussion by RIvance** [carbon-language/carbon-lang#1500] [Language design]
Returning values from block expressions is a useful feature in rust which makes the code cleaner. 
It might be a good idea to introduce a similar syntax. 

For example:

```
var fruit_name: String = "";
if (fruit.IsYellow()) {
  fruit_name = "Banana!";
} else if (fruit.IsOrange()) {
  fruit_name = "Orange!";
} else {
  fruit_name = "Vegetable!";
}
SomeFunction(fruit_name);
```

Can be simplified as:

```
let fruit_name: auto = if (fruit.IsYellow()) {
  "Banana!"
} else if (fruit.IsOrange()) {
  "Orange!"
} else {
  "Vegetable!"
}
SomeFunction(fruit_name);
```

Or we can use a keyword `eval` to make the syntax more elegant:

```
let fruit_name: auto = eval if (fruit.IsYellow()) {
  break "Banana!";
} else if (fruit.IsOrange()) {
  break "Orange!";
} else {
  break "Vegetable!";
}
SomeFunction(fruit_name);
```

The function of `break` here is to return the following variable (or expression) as the value of the block expression

**Comment by jwtowner** [comment]
Looks like there was a discussion on this here: https://github.com/carbon-language/carbon-lang/issues/430

**Comment by YagaoDirac** [comment]
I'm sorry, I disagree.
They are almost the same, on both readability and write-ability. But to do so, more grammar is needed. Also, new grammar is not very regular which is hard for auto completion to handle.
If you really like this grammar, I think a better way is to do it like:
{
scope.enable_rust_block_expression(true);
//let me steal your second part
let fruit_name: auto = if (fruit.IsYellow()) {
  "Banana!"
} else if (fruit.IsOrange()) {
  "Orange!"
} else {
  "Vegetable!"
}
SomeFunction(fruit_name);
}
}
And then, for the third part:
{
scope.enable_RIvance_eval_grammar(true);
//then I'm gonna steal your third part
...
}
